csodiaq/csodiaqParser.py

```diff
@@ -0,0 +1,329 @@
+    )
+
+
+def add_reanalysis_parser(commandParser):
+    reanalysisParser = commandParser.add_parser(
+        "targetedReanalysis",
+        help="Creates files readable by a mass spectrometer for targeted reanalysis of identified peptides.",
+    )
+    reanalysisParser.add_argument(
+        "-i",
+        "--input",
+        type=ScoringOutputDirectory(),
+        required=True,
+        help="Directory that contains outputs of the scoring step of CsoDIAq.\nRequired.",
+    )
+    reanalysisParser.add_argument(
+        "-p",
+        "--protein",
+        type=RestrictedInt("protein", minValue=1),
+        default=0,
+        help="Determines the maximum number of peptides per identified protein to include.\nOptional. Not setting this variable will result in evaluating peptides only with no reference to proteins.",
+    )
+    reanalysisParser.add_argument(
+        "-heavy",
+        "--heavyIsotope",
+        default=False,
+        action="store_true",
+        help="This flag indicates that files for targeted re-analysis should include heavy fragment isotopes for SILAC quantification.\nOptional.",
+    )
+    reanalysisParser.add_argument(
+        "-b",
+        "--binValueProximity",
+        type=RestrictedFloat("binValueProximity", minValue=0.01),
+        default=0.75,
+        help="When setting bin values, this option indicates how close an m/z value must be to the bin value. Default is 0.75.\nOptional.\nNOTE: Multiple targeted m/z values may fall within a range that a mass spectrometer can identify in one scan. Thus, m/z values are binned to prevent redundant reanalysis.\nExample: let's say we have the m/z values of 199.5 and 200.5, a binValueProximity value of 0.75, and a bin value of 200.0.\nBoth of these m/z values would be in the same bin, as they are both with 0.75 of 200.0.",
+    )
+
+
+def check_for_conflicting_args(args):
+    if args["command"] == "id" and args["histogram"] and args["noCorrection"]:
+        raise argparse.ArgumentTypeError(
+            "The histogram flag is invalidated by the noCorrection flag. Please inspect your input and remove one of the tags."
+        )
+    if args["command"] == "id" and args["correctionDegree"] and args["noCorrection"]:
+        raise argparse.ArgumentTypeError(
+            "The correctionDegree parameter is invalidated by the noCorrection flag. Please inspect your input and remove one of them."
+        )
+    if (
+        args["command"] == "targetedReanalysis"
+        and args["protein"]
+        and len(args["input"]["protein"]) == 0
+    ):
+        raise argparse.ArgumentTypeError(
+            "The protein argument requires the presence of protein FDR files to function. Please run the protein scoring workflow or remove the protein argument from your commands."
+        )
+
+
+def create_output_name(commandName):
+    return f"csodiaq-{commandName}-{time.strftime('%Y%m%d-%H%M%S')}"
+
+
+def create_new_output_directory_path(newDirectoryLocation, commandName):
```

Comment:
I would expect `create_new_output_directory_path` to make the directory (i.e., `os.mkdir`). I'd recommend switching to `get_pathname` or something similar.

Reply:
get point, switched "create" to "get" in each case